mini-frame.el

```diff
@@ -310,15 +312,24 @@ ALIST is passed to `window--display-buffer'."
       (ignore which-key-popup-type)
       (unwind-protect
           (mini-frame--display fn args)
-        (when (frame-live-p mini-frame-completions-frame)
-          (make-frame-invisible mini-frame-completions-frame))
-        (when (frame-live-p mini-frame-selected-frame)
-          (select-frame-set-input-focus mini-frame-selected-frame))
-        (when (frame-live-p mini-frame-frame)
-          (if (eq system-type 'windows-nt)
-              ;; FIXME sometime buffer is not visible on windows
-              (delete-frame mini-frame-frame)
-            (make-frame-invisible mini-frame-frame))))))))
+        (when (= (recursion-depth) 0)
+          (when (frame-live-p mini-frame-completions-frame)
+            (make-frame-invisible mini-frame-completions-frame)
+            (modify-frame-parameters mini-frame-completions-frame '((parent-frame))))
+          (if (frame-parameter mini-frame-frame 'parent-frame)
```

Comment:
In which case `parent-frame` parameter of `mini-frame` will be `nil` here?

Reply:
As in the description, when I hide the `mini-frame`, I also detach its parent from it.
The purpose of that is to make sure `mini-frame` is not being destroyed when parent frame is gone.
For example, you open another frame `M-x make-frame` or using `emacsclient` to connect to server. At that time, if `mini-frame` is used inside the new frame, and you close the new frame after that, `mini-frame` will be gone with the new frame.
With this change, it's prevented, so we don't need to create new `mini-frame` again

Reply:
I understand that.

But I mean this particular place (line 319). We are only going to hide mini-frame, so it is visible and it's `parent-frame` is not `nil`, right?

Reply:
I see, Since this part is in `unwind` form, I assume anything may happens.
For example, the `mini-frame` is dead, thus the `frame-parent` parameter is returned as `nil`.
In that case it will go to else code bellow and try to find another root frame to set focus to. So we don't accidently lost the input focus in some unreachable place.